tests/prague/eip2935_historical_block_hashes_from_state/test_block_hashes.py

```diff
@@ -315,6 +315,48 @@ def test_block_hashes_history(
     )
 
 
+@pytest.mark.valid_from("Prague")
+@pytest.mark.with_all_call_opcodes
+def test_block_hashes_call_opcodes(
+    blockchain_test: BlockchainTestFiller, pre: Alloc, call_opcode: Op
+):
+    """Test that the call opcodes can be used to call the history contract and get the block hashes."""  # noqa: E501
+    blocks = []
+    blocks.append(Block())
+
+    storage = Storage()
+    return_code_slot = storage.store_next(0x1)
+
+    code = Op.MSTORE(0, 1) + Op.SSTORE(
+        return_code_slot,
+        call_opcode(
+            address=Spec.HISTORY_STORAGE_ADDRESS,
+            args_offset=0,
+            args_size=32,
+            ret_offset=32,
+            ret_size=32,
+        ),
+    )
```

Comment:
Checking the returned value would be hard because we (currently) don't have a way to compare it against the previous block's real hash, but we could store `Op.ISZERO(Op.MLOAD(32))` as a sanity check that the returned value is not zero, which should actually fail on `call_opcode == Op.DELEGATECALL`.

Reply:
I notice since the DELEGATECALL doesnt touch that slot, we can assume it's zero. I want to be sure that the blockhash being zero for other cases like CALL will not give us false positives. Do you think this will be a problem?